mods/ra/maps/soviet-09/soviet09.lua

```diff
@@ -0,0 +1,108 @@
+--[[
+   Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+TruckStops = { TruckStop1, TruckStop2, TruckStop3, TruckStop4, TruckStop5, TruckStop6, TruckStop7, TruckStop8 }
+MissionStartAttackUnits = { StartAttack1tnk1, StartAttack1tnk2, StartAttackArty1, StartAttackArty2, StartAttackArty3 }
+TruckEscape = { TruckEscape1, TruckEscape2, TruckEscape3, TruckEscape4, TruckEscape5, TruckEscapeWest }
+BackupRoute = { TruckEscape2, TruckEscape1, TruckEscapeEast }
+
+MissionStart = function()
+	Utils.Do(TruckStops, function(waypoint)
+		StolenTruck.Move(waypoint.Location)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		Utils.Do(MissionStartAttackUnits, function(actor)
+			actor.AttackMove(DefaultCameraPosition.Location)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(45), function()
+		Media.DisplayMessage("Commander, the truck has stopped at a nearby allied base.\nAllied radio intercepts say the truck has orders to flee the battlefield\nif any Soviet units approach the base.")
+	end)
+
+	Trigger.OnKilled(StolenTruck, function()
+		USSR.MarkCompletedObjective(DestroyTruck)
+	end)
+
+	Trigger.OnKilled(CommandCenter, function()
+		USSR.MarkFailedObjective(DefendCommand)
+	end)
+end
+
+Trigger.OnEnteredProximityTrigger(TruckAlarm.CenterPosition, WDist.FromCells(13), function(actor, triggerflee)
+	if actor.Owner == USSR and not actor.Type == "badr" and not actor.Type == "u2" then
+		Trigger.RemoveProximityTrigger(triggerflee)
+		Media.DisplayMessage("The convoy truck is attempting to escape!")
+		Media.PlaySoundNotification(USSR, "AlertBleep")
+		Utils.Do(TruckEscape, function(waypoint)
+			StolenTruck.Move(waypoint.Location)
+		end)
+
+		Trigger.OnIdle(StolenTruck, function()
+			Utils.Do(BackupRoute, function(waypoint)
+				StolenTruck.Move(waypoint.Location)
+			end)
+		end)
+	end
+end)
+
+Trigger.OnEnteredFootprint(({ TruckEscapeWest.Location } or { TruckEscapeEast.Location }), function(actor, triggerlose)
+	if actor.Owner == Greece and actor.Type == "truk" then
+		Trigger.RemoveFootprintTrigger(triggerlose)
+		actor.Destroy()
+		USSR.MarkFailedObjective(DestroyTruck)
+	end
+end)
+
+Tick = function()
+	Greece.Cash = 50000
+	Germany.Cash = 50000
+
+	if USSR.HasNoRequiredUnits() then
```

Comment:
This shouldn't be needed since you already lose when you lose the command center, don't you? We can then remove the `BeatUSSR` objective too.